VirtusizeUIKit/Sources/Buttons/VirtusizeUIBaseButton.swift

```diff
@@ -0,0 +1,226 @@
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import Foundation
+
+public class VirtusizeUIBaseButton: UIButton {
+	public var style = VirtusizeUIButtonStyle.default {
+		didSet {
+			setStyle()
+		}
+	}
+
+	internal let rippleView = UIView()
+	internal var rippleShapeLayer: CAShapeLayer?
+	internal var rippleMask: CAShapeLayer {
+		let maskLayer = CAShapeLayer()
+		maskLayer.path = UIBezierPath(
+			roundedRect: bounds,
+			cornerRadius: layer.cornerRadius
+		).cgPath
+		return maskLayer
+	}
+	internal var startTouchLocation: CGPoint?
+	private var _backgroundColor: UIColor?
+
+	public override func beginTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
+		startTouchLocation = touch.location(in: self)
+
+		setRippleLayer(touchLocation: startTouchLocation!, holdAnimation: true)
+
+		UIView.animate(
+			withDuration: 0.1,
+			delay: 0,
+			options: UIView.AnimationOptions.allowUserInteraction,
+			animations: {
+				self.rippleView.alpha = 1
+			},
+			completion: nil
+		)
+
+		return super.beginTracking(touch, with: event)
+	}
+
+	public override func cancelTracking(with event: UIEvent?) {
+		super.cancelTracking(with: event)
+		releaseAnimation()
+	}
+
+	public override func endTracking(_ touch: UITouch?, with event: UIEvent?) {
+		super.endTracking(touch, with: event)
+		releaseAnimation()
+	}
+
+	private func releaseAnimation() {
+		setRippleLayer(touchLocation: startTouchLocation!, holdAnimation: false)
+
+		UIView.animate(withDuration: 0.1, delay: 0, options: UIView.AnimationOptions.allowUserInteraction, animations: {
+			self.rippleView.alpha = 1
+		}, completion: { _ in
+			UIView.animate(withDuration: 0.6, delay: 0, options: UIView.AnimationOptions.allowUserInteraction, animations: {
+				self.rippleView.alpha = 0
+			}, completion: nil)
+		})
+	}
+
+	private func setRippleLayer(touchLocation: CGPoint, holdAnimation: Bool) {
+		if rippleShapeLayer != nil {
+			rippleShapeLayer?.removeFromSuperlayer()
+			rippleShapeLayer = nil
+		}
+		rippleShapeLayer = CAShapeLayer()
```

Comment:
You should avoid using the force unwrap as much as possible `!`.
I think assigning to a local variable for the configuration and then once it is configured assign it back to your property

```swift
	private func setRippleLayer(touchLocation: CGPoint, holdAnimation: Bool) {
		if rippleShapeLayer != nil {
			rippleShapeLayer?.removeFromSuperlayer()
		}
		let shapeLayer = CAShapeLayer()

		let maxBoundSize = max(bounds.size.width, bounds.size.height)
 		shapeLayer.bounds = CGRect(x: 0, y: 0, width: maxBoundSize, height: maxBoundSize)
                ...
                rippleShapeLayer = shapeLayer
```

Reply:
@zr0z yeah, you are right. The changes are on commit 674e9d6